### PR TITLE
docs(orders): DEVDOCS-4469: support shipping consignment in PUT.

### PR DESCRIPTION
**Pull request by bc-munjal**
# [DEVDOCS-4469]

## What changed?
* Support for shipping consignment in PUT.

## Anything else?
Related PRs: https://github.com/bigcommerce/bigcommerce/pull/48811

[DEVDOCS-4469]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4469?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/orders.v2.oas2.yml [hunk 1]
```diff
@@ -99,8 +99,28 @@ paths:
...
+
+        or
 
+        Consignment mode: include the `consignments` array
+        
+        **Usage notes**
+
+        These two methods can't be combined, for instance:
+        - If the `consignments` array is present in the payload, then you can't include `shipping_addresses` nor `products`.
+        - If either `shipping_addresses` or `products` are present, then you can't include the `consignments` array.
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:120]: :clap thanks for adding this Munjal, we missed that before 😄

File: reference/orders.v2.oas2.yml [hunk 2]
```diff
@@ -182,6 +202,35 @@ paths:
...
+                        first_name: "John"
+                        last_name: "Doe"
+                        company: "BC"
+                        street_1: "12 Foo St"
+                        city: "Austin"
+                        state: "Texas"
+                        zip: "78751"
+                        country: "United States"
+                        country_iso": "US"
+                        email: "john.doe@bigcommerce.com"
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:233]: there is no "billing_address" in consignments
also the "shipping" should be an array

File: reference/orders.v2.oas2.yml [hunk 3]
```diff
@@ -3421,6 +3470,7 @@ components:
       title: order_Resp
       description: Order object returned in responses.
       allOf:
+        -
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:3473]: seems like this line is not required

File: reference/orders.v2.oas2.yml [hunk 4]
```diff
@@ -4788,9 +4838,7 @@ components:
       description: |
        **Usage notes:**
 
-       To `add` a custom product to an existing order, don't include `id` in the payload. You must provide a nonempty value for at least one of these fields: `name`, `name_customer`, or `name_merchant`.
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:4791]: why do we need to remove this line ?

Comment (donald-nguyen-bc) [reply reference/orders.v2.oas2.yml:4791]: ah got it, we have `orderCustomProduct_Post` as well, thanks for cleaning this up 🙇

File: reference/orders.v2.oas2.yml [hunk 5]
```diff
@@ -5189,6 +5234,56 @@ components:
           shipping_quotes:
             url: 'https://api.bigcommerce.com/stores/{$$.env.store_hash}/v2/orders/163/consignments/shipping/64/shipping_quotes'
             resource: /orders/163/consignments/shipping/64/shipping_quotes
+    shippingConsignment_Put:
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:5237]: this object schema seems not to be used anywhere ?

File: reference/orders.v2.oas2.yml [hunk 6]
```diff
@@ -5189,6 +5234,56 @@ components:
...
+                anyOf:
+                 - $ref: '#/components/schemas/orderCatalogProduct_Put'
+                 - $ref: '#/components/schemas/orderCatalogProduct_Post'
+                 - $ref: '#/components/schemas/orderCustomProduct_Post'
+                 - $ref: '#/components/schemas/orderCustomProduct_Put'
+      x-examples:
+        example-1:
+          line_items:
+            - id: 12
+              quantity: 2
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:5264]: is this example for `shippingConsignment_Put` ? if it is then "id" is required

**Comment by bc-tgomez** [comment]
Hey @donald-nguyen-bc --just wanting to be sure, is this part of [DEVDOCS-4469](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4469)?

**Comment by donald-nguyen-bc** [comment]
> Hey @donald-nguyen-bc --just wanting to be sure, is this part of [DEVDOCS-4469](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4469)?

yes @bc-tgomez, it should be DEVDOCS-4469.  @bc-munjal, may you pls update the PR description to refer to DEVDOCS-4469 ?

File: reference/orders.v2.oas2.yml (hunk 2 above)
Comment (bc-munjal) [reply reference/orders.v2.oas2.yml]: Thanks for the call out, fixed this ✅

File: reference/orders.v2.oas2.yml (hunk 3 above)
Comment (bc-munjal) [reply reference/orders.v2.oas2.yml]: Yep, got rid of this.

File: reference/orders.v2.oas2.yml (hunk 5 above)
Comment (bc-munjal) [reply reference/orders.v2.oas2.yml:5225]: Added this to `orderConsignment_Put`

File: reference/orders.v2.oas2.yml (hunk 6 above)
Comment (bc-munjal) [reply reference/orders.v2.oas2.yml]: Got rid of this example, as it was not being used, the example values are instead sourced from orderCatalogProduct or orderCustomProduct schemas

File: reference/orders.v2.oas2.yml [hunk 7]
```diff
@@ -5189,6 +5222,29 @@ components:
...
+              type: array
+              minItems: 1
+              items:
+               type: array
+               items:
+                anyOf:
+                 - $ref: '#/components/schemas/orderCatalogProduct_Put'
+                 - $ref: '#/components/schemas/orderCatalogProduct_Post'
+                 - $ref: '#/components/schemas/orderCustomProduct_Post'
+                 - $ref: '#/components/schemas/orderCustomProduct_Put'
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:5247]: do we also support updating other address properties like first_name, last_name, etc ... ?

Comment (bc-munjal) [reply reference/orders.v2.oas2.yml:5247]: yes we do, I think my idea was to put only the required fields here.

Comment (donald-nguyen-bc) [reply reference/orders.v2.oas2.yml:5247]: I think we would need to add all the fields that are supported in the request payload. Maybe we can have a `shippingConsignment_base` and the `shippingConsignment_Put` and `shippingConsignment_Post` can include it inside ?

File: reference/orders.v2.oas2.yml [hunk 8]
```diff
@@ -5189,6 +5222,30 @@ components:
...
+      allOf:
+        - type: object
+          properties:
+            id:
+              type: integer
+              example: 99
+              description: ID of shipping consignment to update.
+          required:
+            - id
+        - $ref: '#/components/schemas/shippingConsignment_Base'
```
Comment (bc-munjal) [inline reference/orders.v2.oas2.yml:5235]: @donald-nguyen-bc : added the shipping consignment base reference here.

File: reference/orders.v2.oas2.yml (hunk 7 above)
Comment (bc-munjal) [reply reference/orders.v2.oas2.yml]: Utilised an already existing `shippingConsignment_base` [here](https://github.com/bigcommerce/api-specs/pull/1017/commits/9a59d94d16cfee5c3826c74aede05723fa464a26#r1012376991)

File: reference/orders.v2.oas2.yml (hunk 8 above)
Comment (donald-nguyen-bc) [reply reference/orders.v2.oas2.yml:5235]: thanks Munjal 🙇

File: reference/orders.v2.oas2.yml [hunk 9]
```diff
@@ -5244,6 +5301,11 @@ components:
           maxItems: 1
           items:
             $ref: '#/components/schemas/digitalConsignment_Put'
+        shipping:
+          type: array
+          minItems: 0
+          items:
+            $ref: '#/components/schemas/shippingConsignment_Put'
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:5308]: maybe not in scope of this PR: in PUT request, we also support adding new consignment as well, so these should be "any of digitalConsignment_Put and digitalConsignment_Post", "any of shippingConsignment_Put and shippingConsignment_Post"

**Review by donald-nguyen-bc — APPROVED** [review]
👏

**Comment by slsriehl** [comment]
disclaimer ^^ the line numbers will have changed after I did more updating / resolution. Please review.

**Comment by donald-nguyen-bc** [comment]
> In the mean time, I've updated @bc-munjal's branch and the ticket branch to be level with the current master branch. I did find one conflict to resolve. @bc-munjal, did you remove line 4478 on purpose? If so, please remove it again. Screenshot included.
> 
> Screen Shot 2022-11-09 at 11 55 52

@slsriehl Munjal is on leave for 3 weeks so I will try go answer all the questions. For this conflict, I don't think Munjal's PR has any changes relating to that block, so please just use the latest from master.

**Review by bc-traciporter — COMMENTED** [review]
I left some editorial comments.

**Pull request merged**